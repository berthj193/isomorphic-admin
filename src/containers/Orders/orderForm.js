import React from 'react';
import PropTypes from 'prop-types';
import { Field } from 'redux-form';
import LineItemInput from './lineItemInput';
import Form, { FormDivider, FormSubmit } from '../../components/uielements/form';
import Select, { SelectOption } from '../../components/uielements/select';
import Input from '../../components/uielements/input';
import { required, email, fieldListRequired } from '../../helpers/formValidation';
import AddressAutocomlete from '../../components/autocomplete/address';
import ShippingRateAutocomlete from '../../components/autocomplete/shippingRate';
import StoreAutocomlete from '../../components/autocomplete/store';

import withReduxForm from '../../components/formWrapper/withReduxForm';

export class OrderForm extends React.Component {
  static propTypes = {
    items: PropTypes.string,
    handleSubmit: PropTypes.func,
    edit: PropTypes.bool,
  };

  static defaultProps = {
    shipping_rates: [],
  };

  render() {
    return (
      <Form onSubmit={this.props.handleSubmit}>
        <Field
          label="Store"
          name="store_key"
          component={StoreAutocomlete}
          validate={[required]}
          required
        />

        <FormDivider />

        <Field
          label="Guest e-mail"
          name="guest_email"
          placeholder="johndoe@isomorphic.com"
          component={Input}
          validate={[email]}
        />
        <Field label="Guest first name" name="guest_given_name" placeholder="John" component={Input} hasFeedback />
        <Field label="Guest last name" name="guest_family_name" placeholder="Doe" component={Input} hasFeedback />
        <Field label="Phone number" name="phone_number" component={Input} hasFeedback />

        <FormDivider />

        <Field label="Currency" name="currency" defaultValue="GBP" component={Select} validate={[required]} required>
          <SelectOption value="GBP">£ GBP</SelectOption>
          <SelectOption value="USD">$ USD</SelectOption>
          <SelectOption value="EUR">€ EUR</SelectOption>
        </Field>

        {
          !this.props.edit
            ? <React.Fragment>
              <Field label="Discount code" name="discount_code" component={Input} hasFeedback />

              <FormDivider />

              <Field
                name="line_items"
                component={LineItemInput}
                validate={[fieldListRequired]}
                required
              />

              <FormDivider />

              <Field
                label="Shipping address"
                name="shipping_address_id"
                component={AddressAutocomlete}
                validate={[required]}
                required
              />
              <Field
                label="Shipping rate"
                name="shipping_rate_id"
                component={ShippingRateAutocomlete}
                validate={[required]}
                required
              />
              <Field label="payment_token" name="payment_token" component={Input} />
            </React.Fragment>
            : null
        }

        <FormSubmit />
      </Form>
    );
  }
}

export default withReduxForm({
  form: 'orderForm',
  enableReinitialize: true,
})(OrderForm);
