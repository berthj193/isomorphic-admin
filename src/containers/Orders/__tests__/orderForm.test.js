import React from 'react';
import { OrderForm } from '../orderForm';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

describe('<OrderForm />', () => {
  const getComponent = (props = {}) => shallow(<OrderForm {...props} />);

  it('should render properly', () => {
    expect(toJson(getComponent())).toMatchSnapshot();
  });
});
