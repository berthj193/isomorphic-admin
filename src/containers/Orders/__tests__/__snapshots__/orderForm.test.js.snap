// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderForm /> should render properly 1`] = `
<Form
  hideRequiredMark={false}
  layout="horizontal"
  onSubmit={[Function]}
  prefixCls="ant-form"
>
  <Field
    component={[Function]}
    label="Store"
    name="store_key"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  />
  <FormDivider />
  <Field
    component={[Function]}
    label="Guest e-mail"
    name="guest_email"
    placeholder="johndoe@isomorphic.com"
    validate={
      Array [
        [Function],
      ]
    }
  />
  <Field
    component={[Function]}
    hasFeedback={true}
    label="Guest first name"
    name="guest_given_name"
    placeholder="John"
  />
  <Field
    component={[Function]}
    hasFeedback={true}
    label="Guest last name"
    name="guest_family_name"
    placeholder="Doe"
  />
  <Field
    component={[Function]}
    hasFeedback={true}
    label="Phone number"
    name="phone_number"
  />
  <FormDivider />
  <Field
    component={[Function]}
    defaultValue="GBP"
    label="Currency"
    name="currency"
    required={true}
    validate={
      Array [
        [Function],
      ]
    }
  >
    <Option
      value="GBP"
    >
      £ GBP
    </Option>
    <Option
      value="USD"
    >
      $ USD
    </Option>
    <Option
      value="EUR"
    >
      € EUR
    </Option>
  </Field>
  <React.Fragment>
    <Field
      component={[Function]}
      hasFeedback={true}
      label="Discount code"
      name="discount_code"
    />
    <FormDivider />
    <Field
      component={[Function]}
      name="line_items"
      required={true}
      validate={
        Array [
          [Function],
        ]
      }
    />
    <FormDivider />
    <Field
      component={[Function]}
      label="Shipping address"
      name="shipping_address_id"
      required={true}
      validate={
        Array [
          [Function],
        ]
      }
    />
    <Field
      component={[Function]}
      label="Shipping rate"
      name="shipping_rate_id"
      required={true}
      validate={
        Array [
          [Function],
        ]
      }
    />
    <Field
      component={[Function]}
      label="payment_token"
      name="payment_token"
    />
  </React.Fragment>
  <FormSubmit />
</Form>
`;
